tests/js/spec/components/autoplayVideo.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import {mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+
+import {AutoplayVideo} from 'sentry/components/autoplayVideo';
+
+jest.mock('react', () => {
+  return {
+    ...jest.requireActual('react'),
+    useRef: jest.fn(),
+  };
+});
+
+// Use a proxy to prevent <video ref={ref}/> from overriding our ref.current
+const makeProxyMock = (video: Partial<HTMLVideoElement>) => {
+  return new Proxy(
+    {current: video},
+    {
+      get(obj, prop) {
+        return obj[prop];
+      },
+      set(obj, prop) {
+        if (prop === 'current') {
+          obj.current = obj.current;
+        }
+        return true;
+      },
+    }
+  );
+};
+
+describe('autoplayVideo', () => {
+  it('sets mute and calls play', () => {
+    const mock = makeProxyMock({
+      muted: false,
+      play: jest.fn().mockReturnValue(Promise.resolve()),
+    });
+
+    // @ts-ignore we are mocking useRef
+    React.useRef.mockImplementation(() => mock);
+
+    const {container} = mountWithTheme(
+      <AutoplayVideo src="https://example.com/video.mp4" />
+    );
+
+    // Videos sadly do not have a role
+    // eslint-disable-next-line
+    expect(container.querySelector('video')).toBeInTheDocument();
```

Comment:
should we update here too?
```suggestion
    expect(screen.getByTestId('autoplay-video')).toBeInTheDocument();
```